packages/manager/modules/vrack/src/dashboard/vrack.controller.js

```diff
@@ -264,82 +267,366 @@ export default class VrackMoveDialogCtrl {
-            (serverInterface) => {
-              allServices.dedicatedServer.push(serverInterface);
-            },
-          );
+  // Get service info for each dedicated cloud retrieved from API
+  updateDedicatedCloudServiceInfo() {
+    return this.$q
+      .all(
+        this.data.eligibleServices.dedicatedCloud.map((serviceId) => {
+          return this.vrackService
+            .getDedicatedCloud(serviceId)
+            .then(({ data }) => {
+              const dedicatedCloud = VrackMoveDialogCtrl.getDedicatedCloudNiceName(
+                data,
+              );
+              return dedicatedCloud;
+            });
+        }),
+      )
+      .then((dedicatedCloud) => {
+        const dedicated = dedicatedCloud.filter(
+          (service) =>
+            service.productReference === DEDICATED_CLOUD_PRODUCT.epcc,
+        );
+        const managedBareMetal = dedicatedCloud.filter(
+          (service) => service.productReference === DEDICATED_CLOUD_PRODUCT.mbm,
+        );
+        this.data.eligibleServices.dedicatedCloud = dedicated;
+        this.data.eligibleServices.managedBareMetal = managedBareMetal;
+      });
+  }
 
-          allServices.dedicatedServerInterface = [];
-        }
+  // Get service info for each dedicated server retrieved from API
+  updateDedicatedServerServiceInfo() {
+    return this.$q
+      .all(
+        this.data.eligibleServices.dedicatedServer.map((serviceId) => {
+          return this.vrackService
+            .getDedicatedServer(serviceId)
+            .then(({ data }) => {
+              const dedicated = VrackMoveDialogCtrl.getDedicatedServerNiceName(
+                data,
+              );
+              return dedicated;
+            });
+        }),
+      )
+      .then((dedicatedServers) => {
+        this.data.eligibleServices.dedicatedServer = dedicatedServers;
+      });
+  }
 
-        if (allServices?.ip?.length > 0) {
-          const ipv6List = allServices.ip.filter((ip) =>
-            this.constructor.isIPv6(ip),
+  updateDedicatedServerInterfaceServiceInfo() {
+    return this.$q
+      .all(
+        this.data.eligibleServices.dedicatedServerInterface.map((service) => {
+          return this.vrackService
+            .getDedicatedServerInterface(service)
+            .then((response) => {
+              let dedicatedServerInterfaceUpdate = service;
+              dedicatedServerInterfaceUpdate.dedicatedServer = response.data;
+              dedicatedServerInterfaceUpdate = VrackMoveDialogCtrl.getDedicatedServerInterfaceNiceName(
+                dedicatedServerInterfaceUpdate,
+              );
+              return dedicatedServerInterfaceUpdate;
+            });
+        }),
+      )
+      .then((dedicatedServerInterfaces) => {
+        // We need to append dedicatedServerInterfaces list to dedicatedServers list.
+        if (dedicatedServerInterfaces.length > 0) {
+          this.data.eligibleServices.dedicatedServer = [];
```

Comment:
**nitpick:** Unnecessary statement as it is overridden in Line 366

Reply:
No I made a concat, not just an overide:
```
this.data.eligibleServices.dedicatedServer = this.data.eligibleServices.dedicatedServer.concat(
            dedicatedServerInterfaces,
          );
```